Content.Client/DisplacementMap/DisplacementMapSystem.cs

```diff
@@ -30,10 +31,10 @@ public bool TryAddDisplacement(DisplacementData data,
             return false;
 
         if (data.ShaderOverride != null)
-            sprite.LayerSetShader(index, data.ShaderOverride);
+            sprite.Comp.LayerSetShader(index, data.ShaderOverride);
 
-        if (sprite.LayerMapTryGet(displacementKey, out var oldIndex))
-            sprite.RemoveLayer(oldIndex);
+        if (_sprite.LayerMapTryGet(sprite.AsNullable(), displacementKey, out var oldIndex, false))
+            _sprite.RemoveLayer(sprite.AsNullable(), oldIndex);
```

Comment:
```suggestion
        _sprite.RemoveLayer(sprite.AsNullable(), displacementKey, false);
```
can be simplified via the overload

Reply:
Nice catch